crates/typst-layout/src/flow/distribute.rs

```diff
@@ -443,7 +472,13 @@ impl<'a, 'b> Distributor<'a, 'b, '_, '_, '_> {
             // the flow, restore the saved checkpoint to move the sticky
             // suffix to the next region.
             if let Some(snapshot) = self.sticky.take() {
-                self.restore(snapshot)
+                // Only restore snapshot if there's no spill
+                // If a sticky breakable element can be spilled to the following page,
+                // it's fine to put some of it on this page, since the next page will still
+                // have the next element on the same page as *some* of the spilled content.
+                if self.composer.work.spill.is_none() {
+                    self.restore(snapshot)
+                }
```

Comment:
Can you add a test where this changes the behavior of existing `sticky: "below"` blocks so we can see what this is actually changing? Want to avoid regressions here.